cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -0,0 +1,336 @@
+"""
+.. _word2vec_and_lda:
+
+Word Embeddings and Topic Modelling with Gensim
+-----------------------------------------------
+
+This example creates six Flyte tasks that:
+
+1. Generate the sample dataset.
+2. Train the word2vec model.
+3. Train the LDA model and display the words per topic.
+4. Compute word similarities.
+5. Compute word movers distance.
+6. Reduce dimensions using t-SNE and generate a plot using FlyteDeck.
+
+"""
+
+# %%
+# First, we import the necessary libraries.
+import logging
+import os
+import random
+import typing
+from dataclasses import dataclass
+from typing import List, Tuple
+
+import flytekit
+import gensim
+import matplotlib.pyplot as plt
+import mpld3
+import nltk
+import numpy as np
+from dataclasses_json import dataclass_json
+from flytekit import Resources, task, workflow
+from flytekit.types.file import FlyteFile
+from gensim import utils
+from gensim.corpora import Dictionary
+from gensim.models import LdaModel, Word2Vec
+from gensim.parsing.preprocessing import STOPWORDS, remove_stopwords
+from gensim.test.utils import datapath
+from nltk.stem import WordNetLemmatizer
+from nltk.tokenize import RegexpTokenizer
+from sklearn.manifold import TSNE
+
+logger = logging.getLogger(__file__)
+
+# %%
+# First, we download the required packages from the nltk library
+# for the preprocessing steps
```

Comment:
```suggestion
# We download the necessary packages from the ``nltk`` library for data preprocessing.
```

@ryankarlos, can we not have this as part of an existing task? How many tasks need this?

Reply:
Only required by first task - i have shifted it to that task